src/shared/dissect-image.c

```diff
@@ -1199,6 +1198,9 @@ static int deferred_remove(DecryptedPartition *p) {
         if (fd < 0)
                 return -errno;
 
+        if (strlen(p->name) > sizeof(dm.name))
+                return -ENAMETOOLONG;
+
```

Comment:
this check definitely makes sense, but I think  we should probably then use plain strcpy() instead of strncpy() here too, after all there's no value in it anymore and in the typical case it does too much.

(but this doesn't really matter much)

Reply:
(thinking about this, maybe we should add:

```c
static inline int strcpy_safe(char *x, size_t n, const char *v) {
        size_t l;
        if (!x || n <= 0 || !v)
                return -EINVAL;
        l = strlen(v);
        if (l > n-1)
                return -ENOBUFS;
        strcpy(x, v);
        return 0;
}
```

Reply:
No, `strcpy` could overflow the buffer with the NUL byte. It's OK if the name is not terminated in the binary structure. `strcpy_safe` seems wrong for the same reason.